src/Controller/ErrorController.php

```diff
@@ -55,4 +55,14 @@ public function beforeRender(Event $event)
 
         $this->viewBuilder()->templatePath('Error');
     }
+
+    /**
+     * afterFilter callback.
+     *
+     * @param \Cake\Event\Event $event Event.
+     * @return void
```

Comment:
I am not so sure about the void part, especially for the other methods here.
If they do (and they can afaik) return a response, e.g. in beforeFilter() execution is bailing early.
IMO "Response|null" would be more correct for callbacks, right?